docs/HowToSupportTizenRtProtectedBuild.md

```diff
@@ -0,0 +1,343 @@
+	....
+	.data : {
+		_sdata = ABSOLUTE(.);
+		....
+		_edata = ABSOLUTE(.);
+	} > ksram
+	/* Here in comparison to ld.script for flat build, 'sram' memory region should be replaced with 'ksram' memory region. */
+}
+```
+
+#### user-space.ld
+- This linker script defines the various 'SECTIONS' in the user binary and indicates their placement in one of the memory regions defined in memory.ld.
+- This script can be created by making a copy of the already created kernel-space.ld script.
+- In this script, declarations of some external functions should be added at the beginning of the script as shown in below snippet. This will be required during linking.
+
+```
+EXTERN(umm_initialize)
+EXTERN(umm_addregion)
+EXTERN(umm_trysemaphore)
+EXTERN(umm_givesemaphore)
+
+EXTERN(malloc)
+EXTERN(realloc)
+EXTERN(zalloc)
+EXTERN(free)
+```
+- Now, the sections assigned to 'kernel' code memory should be replaed with the 'user' code memory and 'kernel' data memory should be replaced with the 'user' data memory.
+- After the build process, when the userspace binary gets created, two additional things are needed to appear as compared to the kernel space binary.
+1. UserSpace Object - This should be placed at the starting of the userspace binary. Add the lines shown in the below snippet before the text section:
+
+```
+.userspace : {
+	*(.userspace)
+} > uflash
+```
+2. A pointer to the user space heap should be placed at the start of the text section. This will be used by the MM(memory manager) module to make allocations. Hence the user heap pointer (.usrheapptr) should be added in the text section of the user memory. Please modify the linker script as follows:
+
+
+```
+.text : {
+	_stext = ABSOLUTE(.);
+	*(.usrheapptr)		/* User heap pointer added in the .text user section */
+	....
+	_etext = ABSOLUTE(.);
+	} > uflash
+```
+
+
+
+### Userspace Initialization
+- The user space data and bss memory regions need to be initialized.
+- For this, a file "CHIPSET_userspace.c" should be created in the folder path <os/arch/arm/src/CHIPSET> and define an API as shown below in the file.
+- A header file "CHIPSET_userspace.h" should also be created at path <os/arch/arm/src/CHIPSET> and declare the below given API.
+- The functionality of this API should be to clear all of user-space .bss and initialize all of user-space .data. This can be done using the address information stored in the MACRO:'USERSPACE' (which is defined using CONFIG_TINYARA_USERSPACE). The bss/data start and end addresses are stored in this macro.
+- Below is an example showing the basic implementation of this API:
+
+```
+void CHIPSET_userspace(void)
+	{
+		  ....
+		/* Clear all of user-space .bss */
+
+		dest = (uint8_t *)USERSPACE->us_bssstart;	/*STORING THE START ADDRESS OF .bss section */
+		end  = (uint8_t *)USERSPACE->us_bssend;		/*STORING THE END ADDRESS OF .bss section */
+
+		while (dest != end)
+		{
```

Comment:
while (...) {

We should not provide the sample with violations of coding rules.
Please check another lines as well.

Reply:
ok